source/agora/node/Ledger.d

```diff
@@ -350,7 +350,20 @@ public class Ledger
             }
         }
 
-        return null;
+        if (data.enrolls.length >= Enrollment.MinValidatorCount)
```

Comment:
I think we also need to add this check to block validation. But we don't want to duplicate the logic.

So I have a suggestion:

Write an overload of `isInvalidReason` in `agora.consensus.validation.Enrollment` which takes an array of Enrollments, the utxo finder, and a count parameter for the number of existing enrollments.

Something like this:

```D
public string isInvalidReason (in Enrollment[] enrolls, UTXOFinder finder, size_t active_enrollments)
{
    if (enrolls.length + active_enrollments < Enrollment.MinValidatorCount)
        return "Enrollment: Insufficient number of active validators";

    foreach (enroll; enrolls)
        if (auto fail_reason = enroll.isInvalidReason(utxo_finder))
            return fail_reason;

    return null;
}
```

Then, use this function from within `validateConsensusData` as well as `isInvalidReason` in `agora.consensus.validation.Block`. Of course this means that `isInvalidReason` in `agora.consensus.validation.Block` will have to be changed to accept another parameter `active_enrollments`. And `validateBlock` in `Ledger` will also have to be changed to provide this argument.

Reply:
> and a count parameter for the number of existing enrollments.

This would be the count of active validators for this next block height that we're about to accept or reject.

Reply:
That's a very good point.
I applied it all.

Reply:
I knew I had to modify the code for the unit test that wasn't using the network.
It's gonna take a while to finish.

Reply:
I solved this problem using the delegator to get the number of active validators at the block height.